src/proto/h1/decode.rs

```diff
@@ -108,7 +109,7 @@ impl Decoder {
                 loop {
                     let mut buf = None;
                     // advances the chunked state
-                    *state = try_ready!(state.step(body, size, &mut buf));
+                    *state = try_ready!(state.step(body, cx, size, &mut buf));
```

Comment:
Likewise here, I'd probably suggest the `cx` be the last argument, since we don't really need to notice it.

Reply:
No problem!

- [ ] move `cx` to be the last argument

Reply:
On second thought, I guess I'll keep this open until https://github.com/rust-lang-nursery/futures-rs/issues/883 is settled?

Reply:
I don't want to burden you with my stylistic whims. If you want to leave things the way they are, I may just shuffle argument orders as I'm touching various files in the future.